multi-cluster-orchestrator/docs/overview.md

```diff
@@ -0,0 +1,262 @@
+clusters for which ClusterProfiles exist, then we filter that using a regex
+matching the ClusterProfile’s namespace/name.
+
+```
+apiVersion: orchestra.multicluster.x-k8s.io/v1alpha1
+kind: MultiKubernetesClusterPlacement
+metadata:
+  name: cluster-list-placement
+spec:
+  rules:
+    - type: "all-clusters"
+    - type: "cluster-name-regex"
+      arguments:
+        regex: "^fleet-cluster-inventory/mco-worker-.*"
+  scaling:
+    autoscaleForCapacity:
+      minClustersBelowCapacityCeiling: 1
+      workloadDetails:
+        namespace: {APP_DEPLOYMENT_NAMESPACE}
+        deploymentName: {APP_DEPLOYMENT_NAME}
+        hpaName: {APP_HPA_NAME}
+```
+
+And here’s another example using a hardcoded list of three eligible clusters on
+which the workload will be autoscaled:
+
+```
+apiVersion: orchestra.multicluster.x-k8s.io/v1alpha1
+kind: MultiKubernetesClusterPlacement
+metadata:
+  name: cluster-list-placement
+spec:
+  rules:
+    - type: "cluster-list"
+      arguments:
+        clusters: "fleet-cluster-inventory/mco-worker-us-west1,fleet-cluster-inventory/mco-worker-us-east1,fleet-cluster-inventory/mco-worker-europe-west4"
+  scaling:
+    autoscaleForCapacity:
+      minClustersBelowCapacityCeiling: 1
+      workloadDetails:
+        namespace: {APP_DEPLOYMENT_NAMESPACE}
+        deploymentName: {APP_DEPLOYMENT_NAME}
+        hpaName: {APP_HPA_NAME}
+```
+
+## MCO Controller
+
+The MCO controller reads the spec of the Placement, makes decisions about
+scheduling and scaling the workload onto clusters, and updates the Placement’s
+status. For a given workload, each cluster will transition through the following
+state machine:
+
+![Diagram placement cluster state machine](images/mco_cluster_state_machine.svg)
+
+Initially, we have a set of eligible clusters as defined by the rules in the
+Placement. The MCO Controller will start by scaling out the workload to the
+first eligible cluster — adding that cluster to the Placement in the ACTIVE
+state.
+
+If a current cluster becomes ineligible to run the workload, it transitions to
+the EVICTING state and the workload is drained and removed from it.
+
+Part of the MCO Controller is the autoscaler which makes scaling decisions based
+on metrics from the clusters (see the Autoscaling section below for details).
+
+If the autoscaler decides that fewer clusters are needed, it moves one of the
+ACTIVE clusters into the SCALING_IN state and we start draining the workload
+from the cluster. Once the workload gets down to one remaining pod, it is
+removed from the cluster entirely.
+
+If the autoscaler decides that more clusters are needed, it first checks if
+there is a SCALING_IN cluster available and if so moves it back to ACTIVE.
+Otherwise it picks the next eligible cluster and adds it to the set of ACTIVE
+clusters.
+
+Health checking is a planned feature but is not currently implemented.
+
+Note: Proactively draining pods from clusters is coming soon but not currently
```

Comment:
If you want an "alert" rather than plain text note: https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts

Reply:
Nice, TIL. Done.